src/arrayMethodShift.js

```diff
@@ -6,7 +6,12 @@
  */
 function applyCustomShift() {
   [].__proto__.shift2 = function() {
-    // write code here
+    const token = this[0];
+    if (this.length === 0) return token;
+    this.reverse();
+    this.length -= 1;
+    this.reverse();
```

Comment:
It is an elegant solution but not optimal. I will approve it, but maybe with rest operator, you can do it simpler and faster?